.travis.yml

```diff
@@ -83,12 +83,23 @@ install:
   - travis_retry sudo luarocks install ldoc
   - travis_retry sudo luarocks install lua-discount
 
-  # Instal luacov-coveralls for code coverage testing.
-  - if [ "$DO_COVERAGE" = "true" ]; then sudo luarocks install luacov-coveralls; else export DO_COVERAGE=0; fi
+  # Install dependencies for code coverage testing.
+  - if [ "$DO_COVERAGE" = "coveralls" ]; then sudo luarocks install luacov-coveralls; fi
+  - if [ "$DO_COVERAGE" = "codecov" ]; then sudo luarocks install luacov-coveralls; fi
+  - if [ "$DO_COVERAGE" = "" ]; then export DO_COVERAGE=0; fi
 
   # Determine custom version.
   - export AWESOME_VERSION="${TRAVIS_BRANCH}-g$(git rev-parse --short HEAD)"
   - 'if [ "$TRAVIS_PULL_REQUEST" != false ]; then AWESOME_VERSION="${AWESOME_VERSION}-PR${TRAVIS_PULL_REQUEST}"; fi'
+  # function for codecov support
+  - |
+    do_codecov() {
+      if [ "$DO_COVERAGE" = "codecov" ]; then
+        (cd build && luacov) || exit 1
+        bash <(curl -s https://codecov.io/bash) -F "$1" || exit 1
```

Comment:
Add `travis_retry` here?
Maybe also fetch the script only once into `/tmp`, and then use it from there?!